sdk/batch/batch-rest/MigrationGuide.md

```diff
@@ -0,0 +1,265 @@
+```
+
+## Error handling
+
+Previously in `@azure/batch`, the client operation succeed only when service return expected HTTP status code, for example `201` for create resources operations or `200` for general HTTP GET requests. Unexpected HTTP status code will throw a `RestError` from `@azure/ms-rest-js` package. The following example demonstrate how to handle different errors might occur in the get pool request.
+
+```typescript
+import { RestError } from "@azure/ms-rest-js";
+
+try {
+  const pool = await client.pool.get("<pool-id>");
+  console.log("Get pool success: ", pool)
+} catch (error) {
+  if (error instanceof RestError) {
+    // Returned HTTP status is not 200
+    console.log(`Service return unexpected status code ${error.statusCode}: ${error.body}`)
+  } else {
+    // Other errors like connection errors or other exceptions
+    console.log("Failed to get pool with error: ", error)
+  }
+}
+```
+
+Now, for `@azure-rest/batch`, the client operation won't throw errors even when the returned HTTP status code is unexpected, instead it exports a helper method `isUnexpected` to help you check if the response is unexpected. The following example demostarte how to handle different errors might occur in the get pool request.
+
+```typescript
+try {
+  const response = await client.path("/pools/{poolId}", "<pool-id>").get();
+  if (isUnexpected(response)) {
+    // Returned HTTP status is not 200
+    console.log(`Service return unexpected status code ${response.status}: ${response.body}`)
+  } {
+    console.log("Get pool success: ", response.body)
+  }
+} catch (error) {
+  // Other errors like connection errors or other exceptions
+  console.log("Failed to get pool with error: ", error)
+}
+```
+
+## More examples
+
+### Create pools
+
+Previously in `@azure/batch`, you can use the `BatchServiceClient` instance to create a pool with the `pool.add` method. The following example demonstrates how to create a pool with the `BatchServiceClient` instance.
+
+```typescript
+import { BatchServiceModels } from "@azure/batch";
+import { RestError } from "@azure/ms-rest-js";
+const poolParams: BatchServiceModels.PoolAddParameter = {
+  id: "<pool-name>",
+  vmSize: "Standard_D1_v2",
+  virtualMachineConfiguration: {
+    nodeAgentSKUId: "batch.node.windows amd64",
+    imageReference: {
+      publisher: "microsoftwindowsserver",
+      offer: "windowsserver",
+      sku: "2022-datacenter",
+    },
+  },
+  networkConfiguration: {
+    enableAcceleratedNetworking: true,
+  },
+  targetDedicatedNodes: 1,
+};
+const result = await client.pool.add(poolParams);
+console.log("Pool created");
+```
+
+Now in `@azure-rest/batch`, you can use the `path` method of the client instance to send a POST request to the `/pools` endpoint with the pool parameters. Note the `CreatePoolParameters` interface has a `body` field to hold the request body and a `contentType` field to specify the content type of the request.
```

Comment:
Would be good to follow up and see if we can remove the requirement to set contentType explicitly here.

Reply:
I will follow up with Jose